jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -558,16 +558,11 @@ ecma_string_to_number (const ecma_string_t *str_p) /**< ecma-string */
 
       ecma_number_t num;
 
-      MEM_DEFINE_LOCAL_ARRAY (str_buffer_p, string_size, lit_utf8_byte_t);
-
-      lit_utf8_size_t bytes_copied = ecma_string_to_utf8_string (str_p,
-                                                                 str_buffer_p,
-                                                                 string_size);
-      JERRY_ASSERT (bytes_copied == string_size);
+      ECMA_STRING_TO_UTF8_STRING (str_p, str_buffer_p);
 
       num = ecma_utf8_string_to_number (str_buffer_p, string_size);
 
-      MEM_FINALIZE_LOCAL_ARRAY (str_buffer_p);
+      ECMA_FINALIZE_UTF8_STRING (str_buffer_p);
```

Comment:
If you go up a bit you can see:

    const lit_utf8_size_t string_size = ecma_string_get_size (str_p);

ECMA_STRING_TO_UTF8_STRING gets this value as well, which should make the getter above unnecessary.